drake/systems/primitives/test/random_source_test.cc

```diff
@@ -33,17 +33,36 @@ GTEST_TEST(TestSignalLogger, GaussianWhiteNoise) {
       0, 0.0);
 
   simulator.Initialize();
-  simulator.StepTo(5);
+  simulator.StepTo(20);
 
   const auto& x = logger->data();
 
-  for (double threshold = -2.0; threshold < 2.0; threshold += 0.5) {
-    double count = (x.array() < threshold).cast<double>().matrix().sum();
-    // Probability of x < threshold for a Gaussian can be computed with erf:
-    EXPECT_NEAR(count / x.size(),
-                .5 + std::erf(threshold / std::sqrt(2.0)) / 2.0,
-                0.02);  // Note intentionally very large tolerance.
-    // TODO(russt): Tighten tolerance once #4325 is resolved.
+  // Cumulative distribution function of the standard normal distribution.
+  auto Phi = [](double z) { return 0.5 * std::erfc(-z / std::sqrt(2.0)); };
+
+  const double h = 0.1;
+  const int N = x.size();
+  // Evaluate all subintervals (a,a+h) in (-2,2).
+  for (double a = -2.0; a <= 2.0 - h; a += h) {
+    // Counts the number of samples in (a,a+h).
+    const double count =
+        (x.array() >= a && x.array() <= a + h).cast<double>().matrix().sum();
+
+    // Basic confidence interval statistics.  See, for instance,
+    //    Simulation and the Monte Carlo Method (Third Edition)
+    //      by Rubinstein and Kroese, 2017
+    //    page 108,
+    // where I've used Y as the indicator function of a ≤ X ≤ a+h,
+    // E[Y] = Phi(a+h)-Phi(a), Var(Y) = E[Y]-E[Y]², since E[Y²]=E[Y],
```

Comment:
Should be `Var(Y) = E[Y^2] - E[Y]^2`

Reply:
The point of my comment was to argue that the form above is correct...?